sql/src/main/java/io/crate/metadata/TransactionContext.java

```diff
@@ -38,15 +39,15 @@ static TransactionContext of(String userName, SearchPath searchPath) {
 
     SearchPath searchPath();
 
+    SessionSettings sessionSettings();
+
     class StaticTransactionContext implements TransactionContext {
 
-        private final String userName;
-        private final SearchPath searchPath;
+        private final SessionSettings sessionSettings;
         private Long currentTimeMillis;
 
-        StaticTransactionContext(String userName, SearchPath searchPath) {
-            this.userName = userName;
-            this.searchPath = searchPath;
+        StaticTransactionContext(SessionSettings sessionSettings) {
```

Comment:
It looks a bit strange that `SessionSettings` itself already implements most of the `TransactionContext` interface. Would it have made sense to make the `TransactionContext´ directly writable instead? Or reduce the redundant properties so that consumers have to go through `sessionSettings()` to get the information?

Reply:
yeap, I was about to trim down the `TransactionContext` interface to include only 
`long currentTimeMillis();`
`SessionSettings sessionSettings();`